compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4658,7 +4658,7 @@ object Types extends TypeUtils {
           case AliasingBounds(alias) if isMatchAlias =>
             trace(i"normalize $this", typr, show = true) {
               MatchTypeTrace.recurseWith(this) {
-                alias.applyIfParameterized(args.map(_.normalized)).tryNormalize
+                alias.applyIfParameterized(args).tryNormalize
```

Comment:
I agree this delays, but it might very well duplicate as well.   For example `type Triple[A] = (A, A, A)` `Triple[Head[h *: tail]]`.normalize.

And there's also the arbitrary "deep subtypes" threshold which could hit if things are nested deeper.

Reply:
I could be mistaken but .. I don't think the above example would reach `args.map(_.normalized)`, since `Triple` is _not_ a match alias, it's just applied an argument which is an applied match alias. This is what I mean by hard to predict.

I completely agree however that the changes can introduce duplicate computation and that it can cause more "deep subtypes" to occur. I'm not sure `args.map(_.normalized)` is the right way to address these issues though.